starboard/nplb/maximum_player_configuration_explorer_test.cc

```diff
@@ -0,0 +1,117 @@
+  constexpr int kResource_types = 3;
+  constexpr int kMaxInstances = 5;
+
+  std::set<std::vector<int>> test_pattern_set = {
+      {1, 4, 5}, {2, 5, 3}, {3, 3, 4}, {3, 4, 3}, {4, 0, 3}, {5, 5, 2}};
+
+  auto test_funtion = std::bind(PosetContainedByMaxElementsSet,
+                                test_pattern_set, std::placeholders::_1);
+
+  auto max_test_config_set = SearchPosetMaximalElementsDFS(
+      kResource_types, kMaxInstances, test_funtion);
+
+  EXPECT_EQ(max_test_config_set, test_pattern_set);
+}
+
+class MaximumPlayerConfigurationExplorerTest
+    : public ::testing::TestWithParam<
+          ::testing::tuple<SbPlayerOutputMode, const char*>> {
+ protected:
+  MaximumPlayerConfigurationExplorerTest() {}
+
+  testing::FakeGraphicsContextProvider fake_graphics_context_provider_;
+};
+
+TEST_P(MaximumPlayerConfigurationExplorerTest, SunnyDay) {
+  const int kMaxPlayerInstancesPerConfig = 5;
+  const int kMaxTotalPlayerInstances = 5;
+
+  SbPlayerOutputMode output_mode = std::get<0>(GetParam());
+  const char* key_system = std::get<1>(GetParam());
+
+  const std::vector<const char*>& video_test_files = GetVideoTestFiles();
+  std::vector<SbPlayerTestConfig> supported_player_configs;
+  for (const auto& video_filename : video_test_files) {
+    VideoDmpReader dmp_reader(video_filename,
+                              VideoDmpReader::kEnableReadOnDemand);
+    SB_DCHECK(dmp_reader.number_of_video_buffers() > 0);
+    if (SbMediaCanPlayMimeAndKeySystem(dmp_reader.video_mime_type().c_str(),
+                                       key_system)) {
+      supported_player_configs.emplace_back(nullptr, video_filename,
+                                            output_mode, key_system);
+    }
+  }
+
+  if (supported_player_configs.empty()) {
+    // TODO: Use GTEST_SKIP when we have a newer version of gtest.
+    SB_LOG(INFO) << "No supported video codec. Skip the tests.";
+    return;
+  }
+
+  MaximumPlayerConfigurationExplorer explorer(
+      supported_player_configs, kMaxPlayerInstancesPerConfig,
+      kMaxTotalPlayerInstances, &fake_graphics_context_provider_);
+  explorer.CalculateMaxTestConfigs();
+}
+
+std::string GetTestConfigName(
+    ::testing::TestParamInfo<::testing::tuple<SbPlayerOutputMode, const char*>>
+        info) {
+  const ::testing::tuple<SbPlayerOutputMode, const char*>& config = info.param;
+
+  SbPlayerOutputMode output_mode = std::get<0>(config);
+  const char* key_system = std::get<1>(config);
+
+  std::string name = FormatString(
+      "output_%s_key_system_%s",
+      output_mode == kSbPlayerOutputModeDecodeToTexture ? "decode_to_texture"
+                                                        : "punch_out",
+      strlen(key_system) > 0 ? key_system : "null");
+  std::replace(name.begin(), name.end(), '.', '_');
+  std::replace(name.begin(), name.end(), '(', '_');
+  std::replace(name.begin(), name.end(), ')', '_');
+  return name;
+}
+
+INSTANTIATE_TEST_CASE_P(MaximumPlayerConfigurationExplorerTests,
```

Comment:
Because you are mixing `TEST` with `TEST_P`, you'll likely need to add `GTEST_ALLOW_UNINSTANTIATED_PARAMETERIZED_TEST(MaximumPlayerConfigurationExplorerTest);` to avoid errors. I just ran into this issue with a different test. 
http://google.github.io/googletest/reference/testing.html#GTEST_ALLOW_UNINSTANTIATED_PARAMETERIZED_TEST

Reply:
TEST is used for SearchPosetMaximalElementsTests, TEST_P is used for MaximumPlayerConfigurationExplorerTest. In that case, does it also need GTEST_ALLOW_UNINSTANTIATED_PARAMETERIZED_TEST?

Reply:
Discussed with Austin offline, it should be fine as they're in two test suites.